### PR TITLE
Update CI config.

### PR DESCRIPTION
**Pull request by benjyw**

### CODE REVIEW

File: .github/workflows/pants.yaml [hunk 1]
```diff
@@ -21,20 +21,21 @@ jobs:
...
       # See: github.com/pantsbuild/actions/tree/main/init-pants/
       with:
         pants-python-version: ${{ matrix.python-version }}
         # cache0 makes it easy to bust the cache if needed
         # just increase the integer to start with a fresh cache
         gha-cache-key: cache0-py${{ matrix.python-version }}
-        # The Python backend uses named_caches for Pip/PEX state,
-        # so it is appropriate to invalidate on lockfile changes.
-        named-caches-hash: ${{ hashFiles('python-default.lock') }}
+        # The JVM backend doesn't use named caches. See the example-python repo for an example
```
Comment (benjyw) [inline .github/workflows/pants.yaml:36]: Is this statement true?

File: .github/workflows/pants.yaml [hunk 2]
```diff
@@ -18,23 +18,24 @@ jobs:
...
+    - uses: pantsbuild/actions/init-pants@v2
       # This action bootstraps pants and manages 2-3 GHA caches.
       # See: github.com/pantsbuild/actions/tree/main/init-pants/
       with:
         pants-python-version: ${{ matrix.python-version }}
         # cache0 makes it easy to bust the cache if needed
         # just increase the integer to start with a fresh cache
         gha-cache-key: cache0-py${{ matrix.python-version }}
-        # The Python backend uses named_caches for Pip/PEX state,
+        # The JVM backend uses named_caches for Coursier state,
```
Comment (benjyw) [inline .github/workflows/pants.yaml:36]: Folks who know the JVM backend, is this correct? Are the contents of the named_caches affected by the JVM lockfile, as they are for Python?

Comment (stuhood) [reply .github/workflows/pants.yaml:36]: That's correct.

The coursier cache _also_ contains the JDK itself, but I don't think that it is worth the complexity of trying to differentiate it.

Comment (benjyw) [reply .github/workflows/pants.yaml:36]: I think that's OK, because that is determined by pants.toml? which is already in the cache key.

**Pull request merged**